### PR TITLE
Dash object detection

### PR DESCRIPTION
**Pull request by ycaokris**
Issue for app: #31 

# App pull request
- [ ] This is a new app
- [x] I am improving an existing app (redesigns/code "makeovers")

## About

- Playground deployment URL (new version): https://dash-playground.plotly.host/dash-object-detection
- Current gallery app URL: (delete this line if inapplicable): https://dash-gallery.plotly.host/dash-object-detection/
~- Python app repository link: (delete this line if you are working on a Python app)~

## Workflow

- [x] I have created a branch in the appropriate monorepo, and the
  elements necessary for successful deployment are in place.
- [x] If the app is a redesigned and/or restyled version of an
  existing gallery app, I've summarized the changes requested in the
  appropriate Streambed issue and confirm that they have been applied.
- [ ] If the app is on the Dash Gallery portal, I have added a link to
  the GitHub repository for the source code in the portal description.
~- [ ] If the app is a reimplementation of a Python gallery app for the
  DashR gallery, the app in this PR mimics, as closely as possible,
  the style and functionality of the existing app.=~
- [x] I have removed *all Google Analytics code* from the app's
  `assets/` folder.

## The pre-review review

I have addressed all of the following questions:

- [x] Does everything in my code serve some purpose? (I have removed
  any dead and/or irrelevant code.)
- [x] Does everything in my code have a clear purpose? (My code is
  readable and, where it isn't, it has been commented appropriately.)]
- [x] Am I reinventing the wheel? (I have used appropriate packages to
  lessen the volume of code that needs to be maintained.)

### CODE REVIEW

**Comment by ycaokris** [comment]
mobile version
screencapture-dash-playground-plotly-host-dash-object-detection-2019-06-27-22_14_07

File: apps/dash-object-detection/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,92 @@
+# Contributing
```
Comment (shammamah) [inline apps/dash-object-detection/CONTRIBUTING.md]: We can probably get rid of this file.

File: apps/dash-object-detection/LICENSE.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+The MIT License (MIT)
```
Comment (shammamah) [inline apps/dash-object-detection/LICENSE.md]: Also get rid of this one (the other apps don't have it)!

**Comment by celinehuang** [comment]
Oops I thought I was taking over this app, but didn't see the pull request so I just pushed the changes instead of adding a review @shammamah just waiting for your review

**Comment by celinehuang** [comment]
💃

File: apps/dash-object-detection/README.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+Try out the [demo app here](https://dash-gallery.plotly.host/dash-object-detection/).
+
+![Animated1](screenshots/dash-object-detection-demo.gif)
+
+## Getting Started
+
+### Using the demo
+
+To get started, select a footage you want to view, and choose the display mode (with or without bounding boxes). Then, you can start playing the video, and the visualization will be displayed depending on the current time.
```
Comment (shammamah) [inline apps/dash-object-detection/README.md]:
```suggestion
To get started, select the footage you want to view, and choose the display mode (with or without bounding boxes). Then, you can start playing the video, and the visualization will be displayed depending on the current time.
```

What does this

>the visualization will be displayed depending on the current time.

mean?

**Comment by ycaokris** [comment]
@shammamah  this should be ✔️ to merge

File: apps/dash-object-detection/README.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+Run the app
+
+```
+python app.py
+```
+
+## About this app
+
+The videos are displayed using a community-maintained Dash video component. It is made by two Plotly community contributors. You can find the [source code here](https://github.com/SkyRatInd/Video-Engine-Dash).
```
Comment (shammamah) [inline apps/dash-object-detection/README.md]: Is this still the correct link? Aren't we using Dash Player?

File: apps/dash-object-detection/app.py [hunk 5]
```diff
@@ -0,0 +1,711 @@
...
+                                    """
+                                ##### What am I looking at?
+                                
+                                This app enhances visualization of objects detected using state-of-the-art Mobile Vision Neural Networks.
+                                Most user generated videos are dynamic and fast-paced, which might be hard to interpret. A confidence
+                                heatmap stays consistent through the video and intuitively displays the model predictions. The pie chart
+                                lets you interpret how the object classes are divided, which is useful when analyzing videos with numerous
+                                and differing objects.
+
+                                ##### More about this dash app
```
Comment (shammamah) [inline apps/dash-object-detection/app.py]:
```suggestion
                                ##### More about this Dash app
```

File: apps/dash-object-detection/app.py [hunk 6]
```diff
@@ -0,0 +1,711 @@
...
+                                
+                                The purpose of this demo is to explore alternative visualization methods for Object Detection. Therefore,
```
Comment (shammamah) [inline apps/dash-object-detection/app.py]:
```suggestion
                                The purpose of this demo is to explore alternative visualization methods for object detection. Therefore,
```

File: apps/dash-object-detection/app.py [hunk 7]
```diff
@@ -0,0 +1,711 @@
...
+                        html.Img(
+                            id="logo-mobile",
+                            src=app.get_asset_url("dash-logo-stripe.png"),
+                        ),
+                        html.Div(
+                            id="header-section",
+                            children=[
+                                html.H4("Object Detection Explorer"),
+                                html.P(
+                                    "To get started, select a footage you want to view, and choose the display mode (with or without"
```
Comment (shammamah) [inline apps/dash-object-detection/app.py]:
```suggestion
                                    "To get started, select the footage you want to view, and choose the display mode (with or without "
```

File: apps/dash-object-detection/app.py [hunk 8]
```diff
@@ -0,0 +1,711 @@
...
+                                    " bounding boxes). Then, you can start playing the video, and the visualization will "
```
Comment (shammamah) [inline apps/dash-object-detection/app.py]:
```suggestion
                                    "bounding boxes). Then, you can start playing the video, and the visualization will "
```

File: apps/dash-object-detection/app.py [hunk 9]
```diff
@@ -0,0 +1,711 @@
...
+                            ],
+                        ),
+                        html.Div(
+                            className="video-outer-container",
+                            children=html.Div(
+                                className="video-container",
+                                children=player.DashPlayer(
+                                    id="video-display",
+                                    style={
+                                        "position": "absolute",
```
Comment (shammamah) [inline apps/dash-object-detection/app.py]: Is it possible to move this out of the component and into a stylesheet?

File: apps/dash-object-detection/README.md (hunk 4 above)
Comment (ycaokris) [reply apps/dash-object-detection/README.md]: This should be directed to the latest Dash player component(https://github.com/plotly/dash-player),source code from community contributor was outdated for over a year and we could quote them in dash-player repo.

**Pull request merged**